src/ORM/Association/ExternalAssociationTrait.php

```diff
@@ -57,7 +57,7 @@ public function foreignKey($key = null)
     {
         if ($key === null) {
             if ($this->_foreignKey === null) {
-                $this->_foreignKey = $this->_modelKey($this->source()->alias());
+                $this->_foreignKey = $this->_modelKey($this->source()->table());
```

Comment:
This might go funny for applications that have foreign keys based on the aliases, and not the classnames.